crates/pathfinder/src/state/sync/l2.rs

```diff
@@ -46,13 +49,72 @@ pub enum Event {
     /// The receiver should return true (if the contract exists) or false (if it does not exist)
     /// for each contract using the [oneshot::channel].
     QueryContractExistance(Vec<ClassHash>, oneshot::Sender<Vec<bool>>),
+    /// A new L2 pending update was polled.
+    Pending(Box<PendingBlock>, Box<sequencer::reply::StateUpdate>),
+}
+
+/// Poll's the Sequencer's pending block and emits [Event::Pending]
+/// until the pending block is no longer connected to our current head.
+///
+/// This disconnect is detected whenever
+/// - `pending.block_hash != head`, or
+/// - `pending` is a fully formed block and not [PendingBlock]
+async fn poll_pending(
```

Comment:
I was expecting this to be done in place of the current "at head sleep". Did I already forget the rationale for not putting this there?

Reply:
Aah nevermind it just need to go to the depths.

Reply:
This is called from there.